api/provider/aws/job_runner.go

```diff
@@ -1,25 +1,256 @@
+	}
+}
+
+func (r *JobRunner) createDeploy(jobID, request string) error {
+	var req models.CreateDeployRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	deploy, err := r.deployProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, deploy.DeployID)
+}
+
+func (r *JobRunner) createEnvironment(jobID, request string) error {
+	var req models.CreateEnvironmentRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	environment, err := r.environmentProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, environment.EnvironmentID)
+}
+
+func (r *JobRunner) createLoadBalancer(jobID, request string) error {
+	var req models.CreateLoadBalancerRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	loadBalancer, err := r.loadBalancerProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, loadBalancer.LoadBalancerID)
+}
+
+func (r *JobRunner) createService(jobID, request string) error {
+	var req models.CreateServiceRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	service, err := r.serviceProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, service.ServiceID)
+}
+
+func (r *JobRunner) createTask(jobID, request string) error {
+	var req models.CreateTaskRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	task, err := r.taskProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, task.TaskID)
+}
+
+func (r *JobRunner) deleteDeploy(jobID, deployID string) error {
+	return r.deployProvider.Delete(deployID)
+}
+
+func (r *JobRunner) deleteEnvironment(jobID, environmentID string) error {
+	loadBalancers, err := r.loadBalancerProvider.List()
+	if err != nil {
+		return err
+	}
+
+	for _, loadBalancer := range loadBalancers {
+		if loadBalancer.EnvironmentID == environmentID {
+			if err := r.deleteLoadBalancer(jobID, loadBalancer.LoadBalancerID); err != nil {
+				return err
+			}
+		}
+	}
+
+	services, err := r.serviceProvider.List()
+	if err != nil {
+		return err
+	}
+
+	for _, service := range services {
+		if service.EnvironmentID == environmentID {
+			if err := r.deleteService(jobID, service.ServiceID); err != nil {
+				return err
+			}
+		}
+	}
+
+	tasks, err := r.taskProvider.List()
+	if err != nil {
+		return err
+	}
+
+	for _, task := range tasks {
+		if task.EnvironmentID == environmentID {
+			if err := r.deleteTask(jobID, task.TaskID); err != nil {
+				return err
+			}
 		}
+	}
 
-		return nil
+	return catchAndRetry(15, func() (shouldRetry bool, err error) {
```

Comment:
Indicates leaky abstraction. Can this retry logic be encapsulated within the environment provider itself?

Reply:
adding changes in a future pr so we don't capture specific errors